dynamic_listings/python/default/computer_science/bfs.py

```diff
@@ -1,17 +1,17 @@
-def traverse_bfs(root):
+def traverse_bfs(root: Node):
     # We will use a queue for this algorithm (simulated by a list)
-    q = []
+    q: list[Node] = []
     # First thing, we enqueue the root
     q.append(root)
     # Now comes the iterative part. This will keep going until
     # the tree is completely explored.
-    while len(q) != 0:
+    while q:
         # We take the first node in the queue
         n = q.pop(0)
         # We enqueue its children, if they exist
-        if n.left is not None:
+        if n.left:
             q.append(n.left)
-        if n.right is not None:
+        if n.right:
             q.append(n.right)
```

Comment:
This is one of those weird situations that make me think using `is not None` may be better. If the Node class had either

- A `__bool__` method that returns "False" for some reason that is not "the absence of a Node" or
- A `__len__` method that returns zero for some reason that is not "the absence of a Node"

These if statements wouldn't work properly, skipping a node that is present.
Usually this change is fine, but it may be problematic as people's codebases become more and more complex.